exchange/exchange.go

```diff
@@ -264,11 +295,11 @@ func (e *exchange) HoldAuction(ctx context.Context, r AuctionRequest, debugLog *
 
 		if targData != nil {
 			// A non-nil auction is only needed if targeting is active. (It is used below this block to extract cache keys)
-			auc = newAuction(adapterBids, len(r.BidRequest.Imp), targData.preferDeals)
+			auc = newAuction(adapterBids, len(r.BidRequestWrapper.BidRequest.Imp), targData.preferDeals)
 			auc.setRoundedPrices(targData.priceGranularity)
 
 			if requestExt.Prebid.SupportDeals {
-				dealErrs := applyDealSupport(r.BidRequest, auc, bidCategory)
+				dealErrs := applyDealSupport(r.BidRequestWrapper.BidRequest, auc, bidCategory)
```

Comment:
Could you write a test to cover this new block of code?

Reply:
This is beyond of the scope of this PR. I'll need to spend time and understand how to test deal support. There are no json tests for this. I would leave it for another PR and avoid to mix it to this PR, it becomes too complicated already.
We can discuss it with team.